static/src/javascripts-legacy/projects/common/modules/experiments/ab.js

```diff
@@ -243,9 +246,10 @@ define([
      */
     function registerCompleteEvent(complete) {
         return function initListener(test) {
-            var variantId = segmentUtil.variantIdFor(test);
 
-            if (segmentUtil.isInTest(test)) {
+            var variantId = getTestVariantId(test.id);
```

Comment:
I don't think we can rely on localStorage to fire the complete event. localStorage is only updated with a test allocation if that test's `canRun` criteria is true (otherwise, it'll be set to `notintest`), but in some cases we need to register the `complete` event on a page after the test treatment, when the `canRun` criteria might be false. That's why we use the MVT ID cookie to register the completion listener: it determines which variant the user would be in without relying on `canRun`.